src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -160,32 +184,38 @@ export class ManifoldSubscriptionCreate {
 
   @Watch('configuredFeatures')
   async fetchCustomCost(configuredFeatures: FeatureMap) {
-    if (!this.connection) {
-      return undefined;
-    }
+    this.removeErrors('cost');
 
     // if not configurable, return plan cost
     if (Object.keys(this.configuredFeatures).length === 0) {
       this.calculatedCost = 0;
       return undefined;
     }
 
+    if (!this.connection) {
+      this.addErrors(dataError('cost', 'cost of selected plan'));
+      throw new Error('Missing property `connection` on `manifold-subscription-create`.');
+    }
+
     // Hide display while calculating
     this.calculatedCost = undefined;
-    this.errors = undefined;
+
+    // If a request is in flight, cancel it
     if (this.controller) {
       this.controller.abort();
     }
 
-    // If a request is in flight, cancel it
     this.controller = new AbortController();
 
-    // TODO pass controller to the request (might need to do this in manifold-init)
     try {
-      const { cost } = await this.connection.gateway.post<
-        { cost: number },
-        { features: FeatureMap }
-      >(
+      interface CostResponse {
```

Comment:
Does it seem weird to define this inline like this, or is that just me?

Reply:
I questioned it at first too, but they're only used once so 🤷‍♂️